File: c#/composite-template/src/dominio/Compuesto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace composite_template.src.dominio
{
    class Compuesto : Componente
    {
        private List<Componente> componentes;

        public Compuesto()
        {
            this.componentes = new List<Componente>();
        }

        public override void operacion()
        {
            this.componentes.ForEach(c => { c.operacion(); });
        }

        public void agregar(params Componente[] componentes)
        {
            this.componentes.AddRange(componentes);
        }

        public void eliminar(Componente componente)
        {
            this.componentes.Remove(componente);
        }
    }
}

File: php/src/app/dominio/Compuesto.php
<?php
namespace dominio;


class Compuesto extends Componente {
    /**
     * @var Componente[]
     */
    private $componentes;

    public function __construct(){
        $this->componentes = [];
    }

    public function operacion(){
        // TODO: Implement operacion() method.
    }

    public function agregar(Componente ... $componentes){
        $this->componentes = array_merge($this->componentes, $componentes);
    }

    public function eliminar(Componente $componente){
        unset($this->componentes[array_search($componente, $this->componentes)]);
    }
}